assignments/prototypes.js

```diff
@@ -1,110 +1,251 @@
-  console.log(swordsman.team); // The Round Table
-  console.log(mage.weapons); // Staff of Shamalama
-  console.log(archer.language); // Elvish
-  console.log(archer.greet()); // Lilith offers a greeting in Elvish.
-  console.log(mage.takeDamage()); // Bruce took damage.
-  console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
 
-  // Stretch task: 
-  // * Create Villain and Hero constructor functions that inherit from the Humanoid constructor function.  
-  // * Give the Hero and Villains different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
-  // * Create two new objects, one a villain and one a hero and fight it out with methods!
\ No newline at end of file
+const mage = new Humanoid({
+	createdAt: new Date(),
+	dimensions: {
+		length: 2,
+		width: 1,
+		height: 1,
+	},
+	healthPoints: 5,
+	name: 'Bruce',
+	team: 'Mage Guild',
+	weapons: [
+		'Staff of Shamalama',
+	],
+	language: 'Common Tongue',
+});
+
+const swordsman = new Humanoid({
+	createdAt: new Date(),
+	dimensions: {
+		length: 2,
+		width: 2,
+		height: 2,
+	},
+	healthPoints: 15,
+	name: 'Sir Mustachio',
+	team: 'The Round Table',
+	weapons: [
+		'Giant Sword',
+		'Shield',
+	],
+	language: 'Common Tongue',
+});
+
+const archer = new Humanoid({
+	createdAt: new Date(),
+	dimensions: {
+		length: 1,
+		width: 2,
+		height: 4,
+	},
+	healthPoints: 10,
+	name: 'Lilith',
+	team: 'Forest Kingdom',
+	weapons: [
+		'Bow',
+		'Dagger',
+	],
+	language: 'Elvish',
+});
+
+console.log(mage.createdAt); // Today's date
+console.log(archer.dimensions); // { length: 1, width: 2, height: 4 }
+console.log(swordsman.healthPoints); // 15
+console.log(mage.name); // Bruce
+console.log(swordsman.team); // The Round Table
+console.log(mage.weapons); // Staff of Shamalama
+console.log(archer.language); // Elvish
+console.log(archer.greet()); // Lilith offers a greeting in Elvish.
+console.log(mage.takeDamage()); // Bruce took damage.
+console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
+
+
+// Stretch task: 
+// * Create Villain and Hero constructor functions that inherit from the Humanoid constructor function.	
+// * Give the Hero and Villains different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
+// * Create two new objects, one a villain and one a hero and fight it out with methods!
+
+
+function Villain(attributes){
+	Humanoid.call(this, attributes);
+};
+
+Villain.prototype = Object.create(Humanoid.prototype);
+
+Villain.prototype.kickOpponent = function(opponent){
+    opponent.healthPoints -= 3;
+    console.log(`${this.name} kicked ${opponent.name} that caused 3 damage.`);
+
+	if(opponent.healthPoints>0) return opponent.takeDamage();
+	return this.killOpponent(opponent);
+};
+
+Villain.prototype.killOpponent = function(opponent){
```

Comment:
Should be moved to the object higher in the inheritance hierarchy to avoid repetitions (the same function is being placed in the second object in the line 184).